spec/models/manageiq/providers/openstack/infra_manager_spec.rb

```diff
@@ -79,4 +79,40 @@
       @ems_cloud.reload.provider.should be_nil
     end
   end
+
+  context "cloud disk usage" do
+    before do
+      @cloud = FactoryGirl.create(:ems_openstack)
+      @infra = FactoryGirl.create(:ems_openstack_infra_with_stack)
+      provider = FactoryGirl.create(:provider)
+      @cloud.provider_id = provider.id
+      @infra.provider_id = provider.id
+      @cloud.save
+      @infra.save
```

Comment:
Leverage ActiveRecord and FactoryGirl here:

```ruby
@cloud = FactoryGirl.create(:ems_openstack)
@infra = FactoryGirl.create(:ems_openstack_infra_with_stack)
FactoryGirl.create(:provider_openstack, :infra_ems => @infra, :cloud_ems => [@cloud])
```

You probably don't have a provider_openstack factory, so you'll have to create one.